internal/observability-operator/helm_test.go

```diff
@@ -0,0 +1,131 @@
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"open-cluster-management.io/addon-framework/pkg/addonfactory"
+	"open-cluster-management.io/addon-framework/pkg/addonmanager/addontesting"
+	"open-cluster-management.io/addon-framework/pkg/agent"
+	addonapiv1alpha1 "open-cluster-management.io/api/addon/v1alpha1"
+	fakeaddon "open-cluster-management.io/api/client/addon/clientset/versioned/fake"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+var (
+	_ = operatorsv1.AddToScheme(scheme.Scheme)
+	_ = operatorsv1alpha1.AddToScheme(scheme.Scheme)
+)
+
+func fakeGetValues(ctx context.Context, k8s client.Client) addonfactory.GetValuesFunc {
+	return func(
+		_ *clusterv1.ManagedCluster,
+		mcAddon *addonapiv1alpha1.ManagedClusterAddOn,
+	) (addonfactory.Values, error) {
+		opts := handlers.BuildOptions(ctx, k8s, mcAddon, addon.ObservabilityOperatorOptions{Enabled: true})
+		oboValues := manifests.BuildValues(opts)
+
+		return addonfactory.JsonStructToValues(oboValues)
+	}
+}
+
+func Test_ObservabilityOperator_AllConfigsTogether_AllResources(t *testing.T) {
+	const oboNamespace = "openshift-cluster-observability-operator"
+	var (
+		// Addon envinronment and registration
+		managedCluster      *clusterv1.ManagedCluster
+		managedClusterAddOn *addonapiv1alpha1.ManagedClusterAddOn
+
+		// Addon configuration
+		addOnDeploymentConfig *addonapiv1alpha1.AddOnDeploymentConfig
+
+		// Test clients
+		fakeKubeClient  client.Client
+		fakeAddonClient *fakeaddon.Clientset
+	)
+
+	// Setup a managed cluster
+	managedCluster = addontesting.NewManagedCluster("cluster-1")
+
+	// Register the addon for the managed cluster
+	managedClusterAddOn = addontesting.NewAddon("test", "cluster-1")
+	managedClusterAddOn.Status.ConfigReferences = []addonapiv1alpha1.ConfigReference{
+		{
+			ConfigGroupResource: addonapiv1alpha1.ConfigGroupResource{
+				Group:    "addon.open-cluster-management.io",
+				Resource: "addondeploymentconfigs",
+			},
+			ConfigReferent: addonapiv1alpha1.ConfigReferent{
+				Namespace: "open-cluster-management-observability",
+				Name:      "multicluster-observability-addon",
+			},
+			DesiredConfig: &addonapiv1alpha1.ConfigSpecHash{
+				ConfigReferent: addonapiv1alpha1.ConfigReferent{
+					Namespace: "open-cluster-management-observability",
+					Name:      "multicluster-observability-addon",
+				},
+				SpecHash: "fake-spec-hash",
+			},
+		},
+	}
+
+	// Setup the fake k8s client
+	fakeKubeClient = fake.NewClientBuilder().
+		WithScheme(scheme.Scheme).
+		Build()
+
+	addOnDeploymentConfig = &addonapiv1alpha1.AddOnDeploymentConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "multicluster-observability-addon",
+			Namespace: "open-cluster-management-observability",
+		},
+		Spec: addonapiv1alpha1.AddOnDeploymentConfigSpec{},
```

Comment:
Here we have to enable COO as if the user had enabled it using the capabilities 
```suggestion
		Spec: addonapiv1alpha1.AddOnDeploymentConfigSpec{
			CustomizedVariables: []addonapiv1alpha1.CustomizedVariable{
				{
					Name:  "observabilityOperator",
					Value: "monitoring.rhobs",
				},
			},
		},

```